awx/api/metadata.py

```diff
@@ -239,7 +239,7 @@ def get_field_info(self, field):
 
         if field.field_name == 'job_type':
             index = 0
-            for choice in res['choices']:
+            for choice in res['choices'].copy():
```

Comment:
py3 doesn't like that we're modifying this data structures _while_ iterating over them.

Reply:
This loop seems problematic to me, even with the copy.  We're popping items out, but always incrementing an index.

Reply:
Yea, I'm a little stumped as to how this _ever_ worked properly.